src/infra/rate-limiting/index.ts

```diff
@@ -126,14 +126,22 @@ const $createRateLimitMiddleware = (
 		ignoreIP = false,
 		allowReset = true,
 	}: { ignoreIP?: boolean; allowReset?: boolean } = {},
-	field?: RateLimitKey,
+	fields?: RateLimitKey,
 ): RateLimitMiddleware => {
 	let fieldFn: RateLimitKeyFn;
-	if (field != null) {
-		if (typeof field === 'function') {
-			fieldFn = field;
+	if (fields != null) {
+		if (typeof fields === 'function') {
+			fieldFn = fields;
+		} else if (Array.isArray(fields)) {
+			fieldFn = (req) =>
+				fields
+					.map((field) => {
+						const path = _.toPath(field);
```

Comment:
This `_.toPath` must occur only once in the setup phase as it is fairly costly when in very hot code (eg in rate limiting code that potentially runs for every request). If we didn't care about pre-calculating the path in order to do it only once then we would instead do `_.get(req, field);` and let lodash do it automatically each time